### PR TITLE
[IDP-1405] Support Forbidden and ForbiddenOfT in TypedResultExtensions

### PR DESCRIPTION
**Pull request by heqianwang**
## Description of changes
Support Forbidden and ForbiddenOfT as a possible response type. Currently, this HttpResult is not present in .NET8.0 and need to be added explicitly.

## Breaking changes
None.

- [x] Updated the documentation of the project to reflect the changes
- [x] Added new tests that cover the code changes

### CODE REVIEW

**Review by PrincessMadMath — APPROVED** [review]

File: src/Shared/HttpResultsStatusCodeTypeHelpers.cs [hunk 1]
```diff
@@ -16,17 +16,19 @@ internal static class HttpResultsStatusCodeTypeHelpers
...
         {"Microsoft.AspNetCore.Http.HttpResults.BadRequest`1", 400},
         {"Microsoft.AspNetCore.Http.HttpResults.UnauthorizedHttpResult", 401},
+        {"Workleap.Extensions.OpenAPI.TypedResult.Forbidden", 403},
+        {"Workleap.Extensions.OpenAPI.TypedResult.Forbidden`1", 403},
         {"Microsoft.AspNetCore.Http.HttpResults.NotFound", 404},
         {"Microsoft.AspNetCore.Http.HttpResults.NotFound`1", 404},
         {"Microsoft.AspNetCore.Http.HttpResults.Conflict", 409},
         {"Microsoft.AspNetCore.Http.HttpResults.Conflict`1", 409},
         {"Microsoft.AspNetCore.Http.HttpResults.UnprocessableEntity", 422},
-        {"Microsoft.AspNetCore.Http.HttpResults.UnprocessableEntity`T", 422},
```
[the remark this answers is not in the document]
Comment (heqianwang) [reply src/Shared/HttpResultsStatusCodeTypeHelpers.cs:24]: Nope, it was not causing any issues with our tests, we could add some tests for each case

File: src/Workleap.Extensions.OpenAPI.Analyzers/EnforceTypedResultsReturnAnalyzer.cs [hunk 2]
```diff
@@ -38,10 +38,11 @@ private sealed class AnalyzerContext(Compilation compilation)
...
         private INamedTypeSymbol? TaskOfTSymbol { get; } = compilation.GetTypeByMetadataName("System.Threading.Tasks.Task`1");
         private INamedTypeSymbol? ValueTaskOfTSymbol { get; } = compilation.GetTypeByMetadataName("System.Threading.Tasks.ValueTask`1");
-        private INamedTypeSymbol? ResultSymbol { get; } = compilation.GetTypeByMetadataName("Microsoft.AspNetCore.Http.IResult");
-        private INamedTypeSymbol? ActionResultSymbol { get; } = compilation.GetTypeByMetadataName("Microsoft.AspNetCore.Mvc.IActionResult");
+        private INamedTypeSymbol? ResultInterfaceSymbol { get; } = compilation.GetTypeByMetadataName("Microsoft.AspNetCore.Http.IResult");
+        private INamedTypeSymbol? ActionResultInterfaceSymbol { get; } = compilation.GetTypeByMetadataName("Microsoft.AspNetCore.Mvc.IActionResult");
+        private INamedTypeSymbol? ActionResultSymbol { get; } = compilation.GetTypeByMetadataName("Microsoft.AspNetCore.Mvc.ActionResult");
 
-        public bool IsValid => this.ActionResultSymbol is not null || this.ResultSymbol is not null;
+        public bool IsValid => this.ActionResultInterfaceSymbol is not null || this.ResultInterfaceSymbol is not null;
```
[the remark this answers is not in the document]
Comment (heqianwang) [reply src/Workleap.Extensions.OpenAPI.Analyzers/EnforceTypedResultsReturnAnalyzer.cs:45]: I attempted to update the name to that, but there are coding standard rules which prevent it:
image

File: src/Workleap.Extensions.OpenAPI.Analyzers/EnforceTypedResultsReturnAnalyzer.cs [hunk 3]
```diff
@@ -38,10 +38,11 @@ private sealed class AnalyzerContext(Compilation compilation)
     {
         private INamedTypeSymbol? TaskOfTSymbol { get; } = compilation.GetTypeByMetadataName("System.Threading.Tasks.Task`1");
         private INamedTypeSymbol? ValueTaskOfTSymbol { get; } = compilation.GetTypeByMetadataName("System.Threading.Tasks.ValueTask`1");
-        private INamedTypeSymbol? ResultSymbol { get; } = compilation.GetTypeByMetadataName("Microsoft.AspNetCore.Http.IResult");
-        private INamedTypeSymbol? ActionResultSymbol { get; } = compilation.GetTypeByMetadataName("Microsoft.AspNetCore.Mvc.IActionResult");
+        private INamedTypeSymbol? ResultInterfaceSymbol { get; } = compilation.GetTypeByMetadataName("Microsoft.AspNetCore.Http.IResult");
```
[the remark this answers is not in the document]
Comment (heqianwang) [reply src/Workleap.Extensions.OpenAPI.Analyzers/EnforceTypedResultsReturnAnalyzer.cs:41]: I attempted to update the name to that, but there are coding standard rules which prevent it:
image

File: src/Workleap.Extensions.OpenAPI.Analyzers/EnforceTypedResultsReturnAnalyzer.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (heqianwang) [reply src/Workleap.Extensions.OpenAPI.Analyzers/EnforceTypedResultsReturnAnalyzer.cs:45]: ActionResult implements IActionResult so it should be present

File: src/Workleap.Extensions.OpenAPI/TypedResult/FobiddenOfT.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    /// Gets the HTTP status code: <see cref="StatusCodes.Status403Forbidden"/>
+    /// </summary>
+    private static int StatusCode => StatusCodes.Status403Forbidden;
+
+    int? IStatusCodeHttpResult.StatusCode => StatusCode;
+
+    private static readonly string[] ContentTypes = new[] { "application/json" };
+
+    /// <inheritdoc/>
+    public Task ExecuteAsync(HttpContext httpContext)
```
[the remark this answers is not in the document]
Comment (heqianwang) [reply src/Workleap.Extensions.OpenAPI/TypedResult/FobiddenOfT.cs:51]: I think adding the logger logic would require duplicating more internal logic from .NET framework; I have no strong opinion regarding this but I would think that it should be a separate PR if we decide to implement that given we would need to add it to InternalServerError as well

**Pull request merged**